File: BLL/MapModels/TopicModel.cs
using System.Collections.Generic;

namespace Business.MapModels
{
    /// <summary>
    /// Represents Topic mapped model.
    /// </summary>
    public class TopicModel : BaseModel
    {
        /// <summary>
        /// Topic's title.
        /// </summary>
        public string Title { get; set; }

        /// <summary>
        /// Topic's title.
        /// </summary>
        public string Text { get; set; }

        /// <summary>
        /// Topic's title.
        /// </summary>
        public int Rating { get; set; }

        /// <summary>
        /// Topic's title.
        /// </summary>
        public bool IsPinned { get; set; }

        /// <summary>
        /// Topic's title.
        /// </summary>
        public bool CommentsAreBlocked { get; set; }

        /// <summary>
        /// Topic's title.
        /// </summary>
        public string PostingDate { get; set; }

        /// <summary>
        /// Topic's title.
        /// </summary>
        public int CommentsCount { get; set; }

        /// <summary>
        /// Topic's title.
        /// </summary>
        public IEnumerable<CommentModel> CommentModels { get; set; }

        /// <summary>
        /// Topic's title.
        /// </summary>
        public int AuthorId { get; set; }

        /// <summary>
        /// Topic's title.
        /// </summary>
        public string AuthorName { get; set; }

        /// <summary>
        /// Topic's title.
        /// </summary>

        public int CommunityId { get; set; }

        /// <summary>
        /// Topic's title.
        /// </summary>
        public string CommunityName { get; set; }
    }
}

File: Forum.WebApi/ForumAngularClient/src/app/models/topic/topic.model.ts
import {CommentModel} from "../comment/comment.model";

export class TopicModel {
  constructor(
    public id?: number,
    public title?: string,
    public text?: string,
    public rating?: number,
    public isPinned?: boolean,
    public commentsAreBlocked?: boolean,
    public postingDate?: string,
    public commentsCount?: number,
    public commentModels?: CommentModel[],
    public authorId?: number,
    public authorName?: string,
    public communityId?: number,
    public communityName?: string,
    public isRatedUp: boolean = false,
    public isRatedDown: boolean = false
  ) {
  }
}
